astra/src/main/java/com/slack/astra/logstore/search/SearchResultUtils.java

```diff
@@ -683,6 +684,26 @@ public static SearchQuery fromSearchRequest(AstraSearch.SearchRequest searchRequ
       }
     }
 
+    AggregatorFactories.Builder aggregatorFactoriesBuilder = null;
+    if (!searchRequest.getAggregationJson().isEmpty()) {
+      SearchModule searchModule = new SearchModule(Settings.EMPTY, List.of());
+      try {
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Object mappers should only be initialized once - consider refactoring this so that these initialization for mappers/parsers only have to be paid one time at class construction.